### DISCUSSION TITLE
Handling outliers with Online Topic Modeling with River

### DISCUSSION THREAD

**Discussion by vantubbe** [MaartenGr/BERTopic#948] [Q&A] [answered]
Personally I've found HDBSCAN produces good results since it places outliers in the -1 Topic. However River lacks this feature. Is there any way to ignore outliers when using River?

**Comment by MaartenGr** [comment, marked as answer]
The difficulty here is with the definition of an outlier. In the context of online clustering, what may be an outlier at timestep *t* might not be at timestep *t+10* when it has seen much more data. Vice versa is also possible. I do believe there are algorithms in River that take outliers into account, like [DenStream](https://riverml.xyz/0.14.0/api/cluster/DenStream/), but I have not used it myself yet. There might need to be some tweaking necessary there in order to generate the `-1` clusters.

**Reply by vantubbe** [reply]
I did not even think of the t+1 vs t+10 scenario, great point.  I'll have a look at DenStream and thank you!
